src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/MSBuildRestoreUtility.cs

```diff
@@ -587,7 +587,7 @@ public static void PersistDGFileIfDebugging(DependencyGraphSpec spec, ILogger lo
                     path = envPath;
                 }
 
-                Directory.CreateDirectory(Path.GetDirectoryName(path));
+                DirectoryUtility.CreateSharedDirectory(Path.GetDirectoryName(path));
```

Comment:
We should only `CreateSharedDirectory` if we use `NuGetFolderPath.Temp` here. If `NUGET_PERSIST_DG_PATH` I think we should keep existing `Directory.CreateDirectory`.